samza-api/src/main/java/org/apache/samza/context/ContainerContext.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.samza.context;
+
+import java.util.Collection;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.metrics.MetricsRegistry;
+
+
+/**
+ * Contains information at container granularity, provided by the Samza framework, to be used to instantiate an
+ * application at runtime.
+ * <p>
+ * Note that application-defined container-level context is accessible through
+ * {@link ApplicationDefinedContainerContext}.
+ */
+public interface ContainerContext {
+  /**
+   * @return processor id for this container
+   */
+  String getProcessorId();
+
+  /**
+   * Returns {@link TaskName}s for all tasks running on this container. There is one {@link TaskName} for each task.
+   * @return {@link TaskName}s for all tasks running on this container
+   */
+  Collection<TaskName> getTaskNames();
```

Comment:
Should this be here in the new API? I think this (and more) could be obtained using JobModel with this processor's ID as the key.

Reply:
Looking at the models, there are a couple of things in JobModel that don't make sense in the public API. One is that it contains a LocalityManager, and the other is the maxChangeLogStreamPartitions field. These are known issues and we'll eventually clean this up. We can consider making them public APIs at that point.

I think it's fine to expose this information in the appropriate context too. If we do that, I think for this particular case it'll be better to expose a ContainerModel equivalent (i.e., Map<TaskName, List<SSPs for TaskName>>) rather than just the task names.

Reply:
Synced offline. We will move the current ContainerModel and TaskModel to the API layer. We will not expose JobModel for now because it needs to be cleaned up (e.g. remove localityManager), and Samza currently does not expose the full job model now anyways.